models/models.go

```diff
@@ -0,0 +1,175 @@
+package models
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// Job - A structure for an import job
+type Job struct {
+	Recipe        string         `json:"recipe"`
+	State         string         `json:"state"`
+	UploadedFileS []UploadedFile `json:"files"`
+}
+
+// NewJob - The requested structure to create a new job
+type NewJob struct {
+	Recipe            string `json:"recipe"`
+	NumberOfInstances int    `json:"NumberOfInstances"`
+}
+
+// Validate - Validate the content of the structure
+func (i NewJob) Validate() error {
+	if i.Recipe == "" || i.NumberOfInstances == 0 {
+		return fmt.Errorf("Missing properties to create import job struct")
+	}
+	return nil
+}
+
+// JobState - The requested structure to update a jobs state
+type JobState struct {
+	State string `json:"state"`
+}
+
+// Validate - Validate the content of the structure
+func (j JobState) Validate() error {
+	if j.State == "" {
+		return fmt.Errorf("No state provided")
+	}
+	return nil
+}
+
+// Event - A structure for an event for an instance
+type Event struct {
+	Type          string `json:"type"`
+	Time          string `json:"time"`
```

Comment:
The only thing stopping me is that a marshal and unmarshal functions are needed for the time structure. I spend some time looking into this